src/components/Header.jsx

```diff
@@ -1,33 +1,53 @@
 "use client"
-import React from 'react'
+import React, { useState, useEffect} from 'react'
 import '../styles/Header.css'
 import Image from 'next/image'
 import homeLogo from "@/assets/homeLogo.svg"
 import Link from 'next/link'
-import { useRouter } from 'next/navigation'
 import HeaderMiddleLink from './HeaderMiddleLink'
+import diamondBlue from '../assets/diamondBlue.svg'
+import coinGold from '../assets/coinGold.svg'
+import avatar from '../assets/avatar.svg'
+import status from '../assets/status.svg'
 
 const Header = () => {
-        
-  return (
-    <>
-        <div className='header'>
-            <Link className='headerLeft' href="/">
-                <Image src={homeLogo} alt="logoHome" />
-            </Link>              
-            <div className='headerMiddle'>
-                <HeaderMiddleLink icon={'🚀'} text={'Missions'} path={'/missions'}/>
-                <HeaderMiddleLink icon={'🏆'} text={'Leaderboard'} path={'/leaderboard'}/>
-                <HeaderMiddleLink icon={'💰'} text={'Rewards'} path={'/rewards'}/>
+    const [flag, setFlag] = useState(false);    
+
+    return (
+        <>
+            <div className='header'>
+                <Link className='headerLeft' href="/">
+                    <Image src={homeLogo} alt="logoHome" />
+                </Link>     
+                {/* <div className="frame">          */}
+                    <div className='headerMiddle'>
+                        <HeaderMiddleLink icon='🚀' text='Missions' path='/missions' onClick={() => setFlag(currentState => !currentState)} flag={flag}/>
+                        <HeaderMiddleLink icon='🏆' text='Leaderboard' path='/leaderboard' onClick={() => setFlag(currentState => !currentState)} flag={flag}/>
+                        <HeaderMiddleLink icon='💰' text='Rewards' path='/rewards' onClick={() => setFlag(currentState => !currentState)} flag={flag}/>
```

Comment:
create constants folder and within it create a file called routes.js

In this file

```
const PAGES = [
  {
    name: 'Quests',
    link: ROUTES.HOME,
    icon: <HomeIcon />,
    id: 'page_1',
  },
  {
    name: 'Leaderboard',
    link: ROUTES.LEADERBOARDS,
    icon: <LeaderboardIcon />,
    id: 'page_2',
  },
  {
    name: 'Rewards',
    link: ROUTES.REWARDS,
    icon: <RewardsIcon />,
    id: 'page_3',
  },
];

const renderMenuItems = () => {
return (
PAGES.map(page => <HeaderMiddleLink {...page} clickHandler={handlePageRoute} />)
)
}

```